File: client/hrms-client/src/services/JobAdvertService.js
import axios from "axios"

const url = "http://localhost:4000/api/jobadverts/"


const getAll = async ()=>{
      const res = await axios.get(url+"getall")

      return res.data
}

const getById = async(id) =>{
      const res = await axios.get(url+"getById?id="+id)

      return res.data
}

const getByActiveAndEmployer = async (active,id)=>{
      const res = await axios.get(url+"getbyactiveandemployer?active="+active+"&id="+id)

      return res.data
}

const getByEmployer = async (id)=>{
      const res = await axios.get(url+"getbyemployer?&id="+id)

      return res.data
}

const getByActived = async (active)=>{
      const res = await axios.get(url+"getbyactived?active="+active)

      return res.data
}

const getByDate = async ()=>{
      const res = await axios.get(url+"getbydate")

      return res.data
}

const add = async (job_advert)=>{
      const res = await axios.post(url+"add",job_advert)

      return res.data
}

const deleteById = async (id)=>{
      const res = await axios.post(url+"deleteById?id="+id)

      return res.data
}

const changeActive = async (id,active)=>{
      const res = await axios.post(url+"changeActive?active="+active+"&id="+id)

      return res.data
}


const service = {getAll,getById,changeActive,getByActiveAndEmployer,deleteById,getByEmployer,getByActived,getByDate,add}

export default service

File: server/hrms/src/main/java/kodlamaio/hrms/business/abstracts/JobAdvertService.java
package kodlamaio.hrms.business.abstracts;

import kodlamaio.hrms.core.utilities.results.Result;
import kodlamaio.hrms.entities.dtos.JobAdvertDto;

public interface JobAdvertService {
    Result getall();
    Result getById(int id);
    Result add(JobAdvertDto jobAdvert);
    Result getByActived(boolean isActived);
    Result getByDate();
    Result getByActiveAndEmployer(boolean active,int id);
    Result getByEmployer(int id);
    Result deleteById(int id);
    Result changeActive(int id,boolean active);
}
